File: src/main/java/com/namu/skill/model/Skill.java
package com.namu.skill.model;

import org.bukkit.entity.Player;

public abstract class Skill {

    public static double power;
    public static double x;
    public static double y;
    public static double z;

    abstract void animation(Player player);

    public void use(Player player) {
        animation(player);
    };

}

File: src/main/java/com/namu/skill/model/Skill.kt
package com.namu.skill.model

import org.bukkit.entity.Entity
import org.bukkit.entity.LivingEntity
import org.bukkit.inventory.ItemStack

abstract class Skill {

    abstract val name: String

    abstract val damage: Double

    abstract val icon: ItemStack

    abstract fun animation(caster: Entity)

    fun use(caster: Entity) {
        animation(caster)
    }

    fun damage(caster: Entity, entity: Entity) {
        entity.getNearbyEntities(0.1, 0.1, 0.1)
                .filterIsInstance<LivingEntity>()
                .forEach { livingEntity -> livingEntity.damage(damage, caster) }
    }

}